hal/mbed_ticker_api.c

```diff
@@ -74,15 +74,17 @@ void ticker_insert_event(const ticker_data_t *const data, ticker_event_t *obj, t
         prev = p;
         p = p->next;
     }
+    
+    /* if we're at the end p will be NULL, which is correct */
+    obj->next = p;
```

Comment:
This change is not needed if `us_ticker_set_interrupt` does nothing more than set the interrupt. 
Anyway, it shouldn't hurt.